packages/portal-proto/src/components/Modals/SetModals/SaveSelectionModal.tsx

```diff
@@ -62,43 +64,31 @@ const SaveSelectionAsSetModal: React.FC<SaveSelectionAsSetModalProps> = ({
   });
 
   useEffect(() => {
-    if (response.isSuccess) {
-      dispatch(
-        addSet({
-          setType,
-          setName: form.values.name.trim(),
-          setId: response.data as string,
-        }),
-      );
-      showNotification({
-        message: "Set has been saved.",
-        closeButtonProps: { "aria-label": "Close notification" },
-      });
-      closeModal();
-    } else if (response.isError) {
-      showNotification({
-        message: "Problem saving set.",
-        color: "red",
-        closeButtonProps: { "aria-label": "Close notification" },
-      });
+    if (opened) {
+      form.setValues((prev) => ({ ...prev, name: initialSetName, top: max }));
     }
-  }, [
-    response.isSuccess,
-    response.isError,
-    response.data,
-    dispatch,
-    setType,
-    closeModal,
-    form.values.name,
-  ]);
+    // form keeps on changing
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [opened, initialSetName, max]);
```

Comment:
Could you try wrapping this: `form.setValues((prev) => ({ ...prev, name: initialSetName, top: max }));` in a memoized callback and see if that solves the dependency issue here?

Reply:
This is done. Thanks to you 😄